pkg/model/iam.go

```diff
@@ -69,51 +73,104 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		name := b.IAMName(role)
 
 		var iamRole *awstasks.IAMRole
-		{
-			rolePolicy, err := b.buildAWSIAMRolePolicy()
-			if err != nil {
-				return err
-			}
 
-			iamRole = &awstasks.IAMRole{
-				Name:      s(name),
-				Lifecycle: b.Lifecycle,
+		arn := ""
+
+		// Want to use a FeatureFlag in front of this to allow the validation to harden
+		if b.Cluster.Spec.AuthRole != nil && featureflag.CustomRoleSupport.Enabled() {
 
-				RolePolicyDocument: fi.WrapResource(rolePolicy),
-				ExportWithID:       s(strings.ToLower(string(role)) + "s"),
+			roleAsString := string(role)
+
+			if role == kops.InstanceGroupRoleMaster && b.Cluster.Spec.AuthRole.Master != nil {
+				arn = *b.Cluster.Spec.AuthRole.Master
+				glog.Warningf("Custom Role Support is enabled, kops will use %s, for %s role, this is an advanced feature please use with great care", arn, roleAsString)
+			} else if role == kops.InstanceGroupRoleNode && b.Cluster.Spec.AuthRole.Node != nil {
+				arn = *b.Cluster.Spec.AuthRole.Node
+				glog.Warningf("Custom Role Support is enabled, kops will use %s, for %s role, this is an advanced feature please use with great care", arn, roleAsString)
 			}
-			c.AddTask(iamRole)
 
 		}
 
-		{
-			iamPolicy := &iam.IAMPolicyResource{
-				Builder: &iam.IAMPolicyBuilder{
-					Cluster: b.Cluster,
-					Role:    role,
-					Region:  b.Region,
-				},
+		// If we've specified an IAMRoleArn for this cluster role,
+		// do not create a new one
+		if arn != "" {
```

Comment:
> Can we build it identically, but just change the Lifecycle (to validate only)?

Identical name?  We do need a different name/id and need kops to not create the policy.  I am not certain that lifecycle assists with that.  Help me understand more.

The policy is usually the big road block for security groups, that at least I have hit.  You have already reviewed this PR before and never brought up this concern.  Your main concern was that you wanted to get Lifecycle in before this PR.  So I do not understand.

@sp-borja-juncosa @StevenACoffman @VinceMD care to comment on why we need this PR?